scarb/src/bin/scarb/commands/build.rs

```diff
@@ -1,10 +1,12 @@
 use anyhow::Result;
 
+use crate::args::BuildArgs;
 use scarb::core::Config;
 use scarb::ops;
 
 #[tracing::instrument(skip_all, level = "info")]
-pub fn run(config: &Config) -> Result<()> {
+pub fn run(args: BuildArgs, config: &Config) -> Result<()> {
     let ws = ops::read_workspace(config.manifest_path(), config)?;
-    ops::compile(&ws)
+    let profile = args.profile_spec.fetch(&ws)?;
```

Comment:
Wouldn't it make sense if `Config` stored `profile_name` in its state?

On top of that, `Workspace` could also expose `profile_name` as a shortcut.